### PR TITLE
[tplinksmarthome] Fixes for hs220, and made RangeExtender readonly

### PR DESCRIPTION
**Pull request by Hilbrand**
Fixes for HS220 that somehow never got into the master. For some unknown reason some fixes on related to HS220 got never into the pull request at the time, but were available in the market place release. This commit fixes that.

Changed Range Extender switches to read only switches. At this moment it's not known what command to send to the Range Extenders to actively switch them, so it's not implemented. To avoid any confusion for the user this is now reflected by having it as a read only switch

### CODE REVIEW

File: addons/binding/org.openhab.binding.tplinksmarthome.test/pom.xml [hunk 1]
```diff
@@ -1,16 +1,18 @@
-<?xml version="1.0" encoding="UTF-8" standalone="no"?><project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
 
-  <modelVersion>4.0.0</modelVersion>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.tplinksmarthome.test/pom.xml]: This XML should use spaces, can you revert this change?

File: addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/device/DimmerDevice.java [hunk 2]
```diff
@@ -38,24 +38,36 @@
...
+     */
+    private boolean handleBrightnessChannel(String channelId, Connection connection, Command command,
             TPLinkSmartHomeConfiguration configuration) throws IOException {
         HasErrorResponse response = null;
 
         if (command instanceof OnOffType) {
             response = setOnOffState(connection, (OnOffType) command);
-        } else if (command instanceof DecimalType) {
-            DecimalType decimalCommand = (DecimalType) command;
+        } else if (command instanceof PercentType) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/device/DimmerDevice.java]: Is this a breaking change?

File: addons/binding/org.openhab.binding.tplinksmarthome.test/pom.xml (hunk 1 above)
Comment (Hilbrand) [reply addons/binding/org.openhab.binding.tplinksmarthome.test/pom.xml]: I shouldn't have done Ctrl-Shift-F :smile:

File: addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/device/DimmerDevice.java (hunk 2 above)
Comment (Hilbrand) [reply addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/device/DimmerDevice.java]: No, the channel is a `Dimmer`, so it should have been `PercentType` all along. It worked because `DecimalType` is a superclass so it won't complain here.

**Comment by martinvw** [comment]
Thanks!
